Makefile

```diff
@@ -1,53 +1,110 @@
+SYSTEMARCH = $(shell uname -i)
+
+ifeq ($(SYSTEMARCH), x86_64)
+TARGETARCH ?= amd64
+PLATFORM=amd64
+else
+TARGETARCH ?= arm64
+PLATFORM=arm64
+endif
+
+IMAGE_TAG ?= $(CONTAINER):$(APPVERSION)-$(TARGETARCH)
 
 RM = /bin/rm
 CP = /bin/cp
 MKDIR = /bin/mkdir
 
-.PHONY: bin-docker build-docker build check clean docs install release test uninstall
+.PHONY: bin-docker bin-for-docker build build-docker check clean common-build docker docs install multi-stage-docker release set-local set-multi-stage test uninstall
 
 .DEFAULT_GOAL := help
 
 GOFILES := $(wildcard *.go)
 
 all: $(PROGRAM)
 
-check:  ##         This checks the current version of Go installed locally
+check:  ##              This checks the current version of Go installed locally
 	@go version
 
-clean:  ##         This will clean all local build artifacts
+clean:  ##              This will clean all local build artifacts
 	$(info Cleaning project...)
 	@rm -f $(PROGRAM)
 	@rm -rf docs/*
-	docker image prune --force
+	@docker image inspect $(CONTAINER):$(APPVERSION) >/dev/null 2>&1 && docker rmi -f $(CONTAINER):$(APPVERSION) $(CONTAINER):$(DATE) || echo -n ""
 
-docs:   ##          Generate docs
+docs:   ##               Generate docs
 	@rm -rf docs/* && cd hugo && hugo && cd ..
 
-build: $(GOFILES)  ##         Build a local binary using APPVERSION parameter or CI as default
+build: $(PROGRAM) ##              Build a local binary using APPVERSION parameter
+
+$(PROGRAM): $(GOFILES)
 	go build -v -ldflags "-s -w -X main.programVersion=$(APPVERSION)"
 
-bin-docker:  ##    Build a local binary based off of a golang base docker image
+bin-docker:  ##         Build a local binary based off of a golang base docker image
 	sudo docker run --rm -v "$(PWD)":/usr/src/myapp:z -w /usr/src/myapp golang:$(GOVERSION) make APPVERSION=$(APPVERSION) build
 
-build-docker: $(PROGRAM) Dockerfile  ##  Generate a CentOS 7 container with APPVERSION tag, using binary from current environment
-	docker build -f Dockerfile --build-arg VERSION=$(APPVERSION) -t $(CONTAINER):$(APPVERSION) -t $(CONTAINER):$(DATE) .
+bin-for-docker: $(GOFILES)  ##     Build a local binary using APPVERSION parameter or CI as default (to be used in docker image)
+# to be used in docker the built binary needs the CGO_ENABLED=0 option
+	CGO_ENABLED=0 go build -v -ldflags "-s -w -X github.com/CrunchyData/pg_featureserv/conf.setVersion=$(APPVERSION)"
+
+build-common: Dockerfile
+	docker build -f Dockerfile \
+		--target $(BUILDTYPE) \
+		--build-arg VERSION=$(APPVERSION) \
+		--build-arg GOLANG_VERSION=$(GOVERSION) \
+		--build-arg TARGETARCH=$(TARGETARCH) \
+		--build-arg PLATFORM=$(PLATFORM) \
+		--build-arg BASE_REGISTRY=$(BASE_REGISTRY) \
+		--build-arg BASE_IMAGE=$(BASE_IMAGE) \
+		--label vendor="Crunchy Data" \
+		--label url="https://crunchydata.com" \
+		--label release="${APPVERSION}" \
+		--label org.opencontainers.image.vendor="Crunchy Data" \
+		--label os.version="7.7" \
+		-t $(IMAGE_TAG) -t $(CONTAINER):$(DATE) .
```

Comment:
DATE tags gets overwritten by whatever arch is the latest to be build (and presumably pushed, once I'm pushing to 'hub again).